### PR TITLE
Webhook log backend tweaks

### PR DESCRIPTION
**Pull request by LawnGnome**
Here are a few odds and ends that became more obvious needs as I worked on #27179:

* The design Rob and I came up with needs the ability to retrieve _all_ webhook logs, as well as unmatched logs, so that ability has been added to the top level query resolver `webhookLogs()` method. (This was already supported in the store anyway.)
* The middleware would fail to write the correct status code if the handler didn't explicitly set one and didn't send any response bytes. It now defaults to `200`.
* I hadn't added tracking for the request URL, which might be useful, so I added that. I also added the method and version, which means we can render the request like it really looked on the wire. (Spoiler: highlight.js has some really nifty formatting for this.)
* Fixed a bug with requesting the `externalService` on webhook logs that don't have an external service.
* Changed the pagination to use `graphqlutil.ConnectionArgs` because our GraphQL library couldn't unmarshal a JSON number into an `int`. Cool. I feel really good about it.

Rather than splitting these up, I've put them in this PR for easier reviewing, but if anyone wants them in separate PRs I am happy to oblige.

### CODE REVIEW

File: internal/types/webhook_logs.go [hunk 1]
```diff
@@ -15,6 +15,9 @@ type WebhookLog struct {
 }
 
 type WebhookLogMessage struct {
-	Header http.Header
-	Body   []byte
+	Header  http.Header
+	Body    []byte
+	Method  string `json:",omitempty"`
```
Comment (eseliger) [inline internal/types/webhook_logs.go:20]: Is this the same as: 
```suggestion
	Method  string `json:"-"`
```
?

File: cmd/frontend/graphqlbackend/webhook_logs.go [hunk 2]
```diff
@@ -22,24 +22,47 @@ import (
...
 	After      *string
 	OnlyErrors *bool
 	Since      *time.Time
 	Until      *time.Time
 }
 
+// webhookLogsExternalServiceID is used to represent an external service ID,
+// which may be a constant defined below to represent all or unmatched external
+// services.
+type webhookLogsExternalServiceID struct {
```
Comment (eseliger) [inline cmd/frontend/graphqlbackend/webhook_logs.go:35]: Maybe we can also just use a type alias for int

**Review by eseliger — APPROVED** [review]

**Review by mrnugget — APPROVED** [review]

File: internal/types/webhook_logs.go (hunk 1 above)
Comment (LawnGnome) [reply internal/types/webhook_logs.go:20]: It's not: `-` would omit the field entirely, whereas we only want it omitted when the string is empty, hence `omitempty`. (Omitting it means that we don't waste database space in the response messages encoding the keys that will never be used.)

(This behaviour is [described in the documentation for `json.Marshal`](https://pkg.go.dev/encoding/json#Marshal).)

File: cmd/frontend/graphqlbackend/webhook_logs.go (hunk 2 above)
Comment (LawnGnome) [reply cmd/frontend/graphqlbackend/webhook_logs.go:35]: Sorry, I think this is a hint of Rust creeping in here on my part: type aliases in Rust are interchangeable with their aliased type, and I explicitly wanted a new type here that wouldn't be interchangeable with `int` so I could use the type system to be sure that I'd caught all the places where this is used. (Of course, in practice I'd express this as an `enum` in Rust, but that's a separate concern!)

Go, however, doesn't have that problem: `type webhookLogsExternalServiceID int64` still implies type casts as appropriate.

So, in summary, change made. Thanks!
